docs/src/references/c8y-configuration-management.md

```diff
@@ -274,107 +274,125 @@ The HTTP filetransfer feature of the `tedge_agent` provides the service to trans
+  3) external device child1: Uploads configuration file to the thin-edge device with the HTTP filetransfer feature.
   
     HTTP PUT request: `http://<ip address of thin-edge devicee>/tedge/tmpfiles`
     
-    The response from the HTTP filetransfer feature contains a _temporary URL_, that is later used to uniquely access that uploaded file.<br/>
-    Example for a _temporary URL_: `/tedge/tmpfiles/<random string>`
+  4) The response from the HTTP filetransfer feature contains a _temporary URL_, that is later used to uniquely access that uploaded file.<br/>
+     Example for a _temporary URL_: `/tedge/tmpfiles/<random string>`
 
-  * external device child1: notifies the the plugin via MQTT about succeeded upload and the _temporary URL_.
+  5) external device child1: notifies the the plugin via MQTT about succeeded upload and the _temporary URL_.
  
     Topic: `tedge/configuration/res/retrieve/{config type}/{childid}`<br/>
     Example: `tedge/configuration/res/retrieve/bar.conf/child1`<br/>
     Payload: ` <temporary URL> `
        
     TODO: Investigate and decide about topic structure and payload in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307). Unhappy paths also to be considered here.
   
-  * C8Y config plugin: recognizes the MQTT notification about the uploaded file and the temprary-URL, 
-                       and uses the filetransfers local API to obtain the file from the _temprary URL_ to some plugin specific local filesystem path (e.g. `/tmp/c8y-cfg-plugin/<childid>_<cfg-type>`.
-
-    * TODO: Investigate and decide about filetransfer's local API in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307), according to topics below:
- 
-      * local API must avoid that there are two copies of the same file at any point in time
-      * local API must avoid that the file content is transfered/copied from one to another place, when the plugin _obtains_ the file. 
-        Instead just the directory link may change (as it is the case for bash `mv` command).
-      * file is removed from filetransfer's specific folder when the file was successfully obtained to the plugin's path.
-      
-      Reasons for requirements above: Saving disc-space and CPU resources for large files.
-
-    * TO-BE-DEFINED:
-    
-      * Decide which error paths to be considered.
-      * Decide if any timeout to be considerd when waiting for upload notification from external device.
+  6) C8Y config plugin: recognizes the MQTT notification about the uploaded file and the _temprary URL_, 
+                       and downloads the file using HTTP the filetransfer with the _temporary URL_ to some temporary location.
+  
+    Example for HTTP GET request: `http://<ip address of thin-edge device>/<temporary URL>`
 
-  * C8Y config plugin: sends the obtained file to C8Y
+  7) C8Y config plugin: sends the downloaded file to C8Y
```

Comment:
Before this step, the plugin needs to delete the config snapshot from the file transfer service's storage with an HTTP DELETE request.

Reply:
HTTP DELETE step is added now.